solidity/src/Base7683.sol

```diff
@@ -0,0 +1,410 @@
+        maxSpent[0] = Output({
+            token: orderData.outputToken,
+            amount: orderData.amountOut,
+            recipient: destinationSettler,
+            chainId: orderData.destinationDomain
+        });
+
+        // this can be used by the filler know how much it can expect to receive
+        Output[] memory minReceived = new Output[](1);
+        minReceived[0] = Output({
+            token: orderData.inputToken,
+            amount: orderData.amountIn,
+            recipient: bytes32(0),
+            chainId: orderData.originDomain
+        });
+
+        // this can be user by the filler to know how to fill the order
+        FillInstruction[] memory fillInstructions = new FillInstruction[](1);
+        fillInstructions[0] = FillInstruction({
+            destinationChainId: orderData.destinationDomain,
+            destinationSettler: destinationSettler,
+            originData: OrderEncoder.encode(orderData)
+        });
+
+        resolvedOrder = ResolvedCrossChainOrder({
+            user: _sender,
+            originChainId: _localDomain(),
+            openDeadline: _openDeadline,
+            fillDeadline: _fillDeadline,
+            minReceived: minReceived,
+            maxSpent: maxSpent,
+            fillInstructions: fillInstructions
+        });
+    }
+
+    function _permitTransferFrom(
+        GaslessCrossChainOrder calldata order,
+        bytes calldata signature,
+        address receiver
+    )
+        internal
+    {
+        OrderData memory orderData = OrderEncoder.decode(order.orderData);
+
+        PERMIT2.permitWitnessTransferFrom(
+            ISignatureTransfer.PermitTransferFrom({
+                permitted: ISignatureTransfer.TokenPermissions({
+                    token: TypeCasts.bytes32ToAddress(orderData.inputToken),
+                    amount: orderData.amountIn
+                }),
+                nonce: order.nonce,
+                deadline: order.openDeadline
+            }),
+            ISignatureTransfer.SignatureTransferDetails({ to: receiver, requestedAmount: orderData.amountIn }),
+            order.user,
+            witnessHash(order),
+            witnessTypeString,
+            signature
+        );
+    }
+
+    function _settleOrder(bytes32 _orderId, bytes32 _receiver, uint32 _settlingDomain) internal {
+        OrderData memory orderData = orders[_orderId];
+
+        if (orderData.destinationDomain != _settlingDomain) revert InvalidDomain();
+        if (orderStatus[_orderId] != OrderStatus.OPENED) revert InvalidOrderStatus();
+
+        orderStatus[_orderId] = OrderStatus.SETTLED;
+
+        address receiver = TypeCasts.bytes32ToAddress(_receiver);
+
+        emit Settled(_orderId, receiver);
+
+        IERC20(TypeCasts.bytes32ToAddress(orderData.inputToken)).safeTransfer(
+            receiver, orderData.amountIn
+        );
+    }
+
+    function _refundOrder(bytes32 _orderId, uint32 _refundingDomain) internal {
```

Comment:
Looks like you lost these comments in this commit https://github.com/BootNodeDev/intents-framework/pull/1/commits/b19d784e715a2fa8057a0667ce1ccff476566c84#diff-1f340c87fc93708649831292d06a5d132d190753e1e1990cb03b1455c14a20e0L372